lesson17.py

```diff
@@ -0,0 +1,139 @@
+# Task 1
+
+class Animal:
+
+    def talk(self):
+        pass
+
+
+class Dog(Animal):
+    def talk(self):
+        return "Woof, Woof"
+
+
+class Cat(Animal):
+    def talk(self):
+        return "Meow"
+
+
+def generic_func(instance_class):
+    if isinstance(instance_class, Animal):
+        return instance_class.talk()
+
+
+dog = Dog()
+cat = Cat()
+
+print("Dog say:", generic_func(dog))
+print("Cat say:", generic_func(cat))
+
+
+# Task 2
+
+
+class Author:
+    def __init__(self, name, country, birthday):
+        self.name = name
+        self.country = country
+        self.birthday = birthday
+        self.books = []
+
+    def __repr__(self):
+        return f"Author ({self.name}, {self.country}, {self.birthday})"
+
+    def __str__(self):
+        return
+
+
+class Book:
+    all_existing_books = 0
+
+    def __init__(self, name, year, author):
+        self.name = name
+        self.year = year
+        self.author = author
+
+    all_existing_books += 1

```

Comment:
This should be placed inside `__init__`. Otherwise this line of code would be processed only once (when python interpreter reads the `Book` class)